libs/execution_context/ebpf_native.c

```diff
@@ -38,9 +40,10 @@ typedef struct _ebpf_native_program
 
 typedef struct _ebpf_native_module
 {
+    ebpf_base_object_t base;
     GUID client_id;
     metadata_table_t* table;
-    volatile int32_t reference_count;
+    bool initializing;
```

Comment:
Can a native module be in more than one state at a time? I.e. is initializing + unloading a valid state combination? 

If not, can we define a set of valid states and make the state machine more explicit?

Reply:
I agree a state enum would be better than a bunch of independent bools.

Reply:
Filed Issue #1586 to track this.